pyEpiabm/pyEpiabm/tests/test_func/test_distancing_functional.py

```diff
@@ -0,0 +1,203 @@
+
+    @patch('pyEpiabm.routine.simulation.tqdm', notqdm)
+    @patch('pyEpiabm.output._CsvDictWriter.write', Mock())
+    @patch('os.makedirs', Mock())
+    @patch("pandas.DataFrame.to_csv")
+    @patch("pandas.read_csv")
+    def test_spatial_enhanced_large(self, mock_read, mock_csv):
+        """Social distancing functional test to ensure less people will be
+        susceptible when spatial enhanced susceptibility increases.
+        """
+        mock_read.return_value = pd.DataFrame(self.pop_params)
+
+        # Enable spatial infectious and social distancing
+        pe.routine.Simulation.set_random_seed(seed=30)
+        pe.Parameters.instance().infection_radius = 1.6
+
+        pe.Parameters.instance().intervention_params = self.intervention
+        pop_standard = TestDistancingFunctional.file_simulation(
+            "test_input.csv", self.sim_params, self.file_params,
+            HelperFunc().sweep_list_initialise())
+
+        self.intervention['social_distancing'][
+            'distancing_spatial_enhanced_susc'] = 0.8
+        pop = TestDistancingFunctional.file_simulation(
+            "test_input.csv", self.sim_params, self.file_params,
+            HelperFunc().sweep_list_initialise())
+
+        mock_read.assert_called_with('test_input.csv')
+        self.assertEqual(mock_csv.call_count, 2)
+
+        # Compare number of susceptible individuals for each age group
+        HelperFunc().assert_greater_equal(
+             pop.cells, pop_standard.cells)
+
+    @patch('pyEpiabm.routine.simulation.tqdm', notqdm)
+    @patch('pyEpiabm.output._CsvDictWriter.write', Mock())
+    @patch('os.makedirs', Mock())
+    @patch("pandas.DataFrame.to_csv")
+    @patch("pandas.read_csv")
+    def test_prob_lower(self, mock_read, mock_csv):
+        """Social distancing functional test to ensure people within the
+        age group of lower enhanced social distancing probability
+        will have less susceptible individuals.
+        """
+        mock_read.return_value = pd.DataFrame(self.pop_params)
+
+        # Enable spatial infectious and social distancing
+        pe.routine.Simulation.set_random_seed(seed=30)
+        pe.Parameters.instance().infection_radius = 1.6
+
+        pe.Parameters.instance().intervention_params = self.intervention
+        pop_standard = TestDistancingFunctional.file_simulation(
+            "test_input.csv", self.sim_params, self.file_params,
+            HelperFunc().sweep_list_initialise())
+
+        self.intervention['social_distancing'][
+            'distancing_enhanced_prob'] = [0.1]*17
+        pop = TestDistancingFunctional.file_simulation(
+            "test_input.csv", self.sim_params, self.file_params,
+            HelperFunc().sweep_list_initialise())
+
+        mock_read.assert_called_with('test_input.csv')
+        self.assertEqual(mock_csv.call_count, 2)
+
+        # Compare number of susceptible individuals for each age group
+        HelperFunc().assert_greater_equal(
```

Comment:
Have you thought about the possibility of tests like this failing by chance? Is there any reason why this has to be the case, or just that it is most likely that these age groups with lower social distancing will have more infections?